keras_nlp/models/bart/bart_seq_2_seq_lm_preprocessor_test.py

```diff
@@ -0,0 +1,161 @@
+            "<mask>": 12,
+        }
+
+        merges = ["Ġ a", "Ġ t", "Ġ k", "Ġ i", "Ġ b", "Ġa i", "p l", "n e"]
+        merges += ["Ġa t", "p o", "r t", "o h", "l i", "Ġi s", "Ġb e", "s t"]
+        merges += ["Ġt h", "Ġai r", "pl a", "Ġk oh", "Ġth e", "Ġbe st", "po rt"]
+        merges += ["pla ne"]
+
+        self.preprocessor = BartSeq2SeqLMPreprocessor(
+            tokenizer=BartTokenizer(
+                vocabulary=vocab,
+                merges=merges,
+            ),
+            encoder_sequence_length=10,
+            decoder_sequence_length=8,
+        )
+
+    def test_tokenize_strings(self):
+        input_data = {
+            "encoder_text": " airplane at airport",
+            "decoder_text": " kohli is the best",
+        }
+
+        output = self.preprocessor(input_data)
+        self.assertAllEqual(
+            output[0]["encoder_token_ids"], [0, 3, 4, 5, 3, 6, 2, 1, 1, 1]
+        )
+        self.assertAllEqual(
+            output[0]["encoder_padding_mask"], [1, 1, 1, 1, 1, 1, 1, 0, 0, 0]
+        )
+        self.assertAllEqual(
+            output[0]["decoder_token_ids"], [0, 7, 8, 9, 10, 11, 2, 1]
+        )
+        self.assertAllEqual(
+            output[0]["decoder_padding_mask"], [1, 1, 1, 1, 1, 1, 1, 0]
+        )
+        self.assertAllEqual(output[1], [7, 8, 9, 10, 11, 2, 1, 1])
+        self.assertAllEqual(output[2], [1, 1, 1, 1, 1, 1, 0, 0])
+
+    def test_tokenize_list_of_strings(self):
+        input_data = {
+            "encoder_text": [" airplane at airport"] * 4,
+            "decoder_text": [" kohli is the best"] * 4,
+        }
+
+        output = self.preprocessor(input_data)
+        self.assertAllEqual(
+            output[0]["encoder_token_ids"], [[0, 3, 4, 5, 3, 6, 2, 1, 1, 1]] * 4
+        )
+        self.assertAllEqual(
+            output[0]["encoder_padding_mask"],
+            [[1, 1, 1, 1, 1, 1, 1, 0, 0, 0]] * 4,
+        )
+        self.assertAllEqual(
+            output[0]["decoder_token_ids"], [[0, 7, 8, 9, 10, 11, 2, 1]] * 4
+        )
+        self.assertAllEqual(
+            output[0]["decoder_padding_mask"], [[1, 1, 1, 1, 1, 1, 1, 0]] * 4
+        )
+        self.assertAllEqual(output[1], [[7, 8, 9, 10, 11, 2, 1, 1]] * 4)
+        self.assertAllEqual(output[2], [[1, 1, 1, 1, 1, 1, 0, 0]] * 4)
+
+    def test_error_multi_segment_input(self):
+        input_data = {
+            "encoder_text": (
+                tf.constant([" airplane at airport"] * 2),
+                tf.constant([" airplane"] * 2),
+            ),
+            "decoder_text": (
+                tf.constant([" kohli is the best"] * 2),
+                tf.constant([" kohli"] * 2),
+            ),
+        }
+
+        with self.assertRaises(ValueError):
+            self.preprocessor(input_data)
+
+    @parameterized.named_parameters(
+        ("tf_format", "tf", "model"),
+        ("keras_format", "keras_v3", "model.keras"),
+    )
+    def test_saved_model(self, save_format, filename):
```

Comment:
two changes we can make for efficiency here...

https://github.com/keras-team/keras-nlp/pull/945 (don't save traces)
https://github.com/keras-team/keras-nlp/pull/894 (mark as large, make separate serialization test)